### PR TITLE
Neilv sprint4

### PR DESCRIPTION
**Pull request by nvohra3**
Pulled Kyle's branch and added features to view and add polling questions for a lecture when in the professor view.

### CODE REVIEW

**Comment by goutamvenkat** [comment]
Site Crashed after I clicked add live lecture and then hit the home button

File: ClassMoodApp/static/tsc/controllers/CreatePollingQuestionController.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                        this.$http = $http;
+                        this.$scope = $scope;
+                        this.$window = $window;
+                        let curLink = $window.location.pathname.split("/");
+                        this.classId = parseInt(curLink[curLink.length-3]);
+                        this.lecture_id = parseInt(curLink[curLink.length-2]);
+                    }
+
+        public submitPollingQuestion(): void {
+		let text = this.$scope.question_text;
```
Comment (goutamvenkat) [inline ClassMoodApp/static/tsc/controllers/CreatePollingQuestionController.ts]: Why is scope being used here? I'm using you are using ng-model and ng-controller for your html. If that's the case, there is no need to use scope. You can instead directly use the instance variables.

File: ClassMoodApp/static/tsc/controllers/CreatePollingQuestionController.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		let a = this.$scope.q.a_text;
+		let b = this.$scope.q.b_text;
+		let c = this.$scope.q.c_text;
+		let d = this.$scope.q.d_text;
+		let answer = this.$scope.q.ans;
+		if (!this.isStudent) {
+			if (text && a && b) {
+				this.$http.get(`/create_polling_question/${this.lecture_id}/${text}/${a}/${b}/${c}/${d}/${answer}`).then(
+					(response: any) => {
+						if (response.data) {
```
Comment (goutamvenkat) [inline ClassMoodApp/static/tsc/controllers/CreatePollingQuestionController.ts]: Preferably use === for strict equality comparison to bools. Javascript is a bit finicky with booleans.

File: ClassMoodApp/static/tsc/controllers/LectureListController.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                            this.$http.get(`/add_lecture/${this.classId}/${lectureName}`).then(
+                                (response: any) => {
+                                    console.log(response.data);
+                                }
+                            )
+                        }
+                    } 
+                }
+            });
+            this.$window.location.reload();
```
Comment (goutamvenkat) [inline ClassMoodApp/static/tsc/controllers/LectureListController.ts]: Do not use reload. Instead make an update function, as in this case, every time u call reload, the entire page has to reload. With an update function, only the table elements will change which will be quicker.

File: ClassMoodApp/templates/createPollingQuestion.html [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{% extends "layout.html" %}
+{% block body %}
+<nav class="navbar navbar-inverse navbar-top">
```
Comment (goutamvenkat) [inline ClassMoodApp/templates/createPollingQuestion.html]: All new html pages will be extending from baseHomePage.html which all ready contains the navbar. This is a page I created, so after merging, ensure that your html is extending from that page, and change the block name to newPage.

File: ClassMoodApp/templates/lectureList.html [hunk 5]
```diff
@@ -0,0 +1,39 @@
+{% extends "layout.html" %}
+{% block body %}
+<nav class="navbar navbar-inverse navbar-top">
```
Comment (goutamvenkat) [inline ClassMoodApp/templates/lectureList.html]: Same thing as createPollingQuestion.html

File: ClassMoodApp/templates/pollingQuestionList.html [hunk 6]
```diff
@@ -0,0 +1,43 @@
+{% extends "layout.html" %}
+{% block body %}
+<nav class="navbar navbar-inverse navbar-top">
```
Comment (goutamvenkat) [inline ClassMoodApp/templates/pollingQuestionList.html]: Same thing as the above html.

**Review by goutamvenkat — CHANGES_REQUESTED** [review]

**Review by krabagobanjo — CHANGES_REQUESTED** [review]
Just realized I didn't comment on this - I didn't find anything to add other than what Goutam found, and most of what he found pertained to my branch, which I changed.  Pulling the latest code from my branch will fix most of these.

Also, what was the error from when the site crashed?  I'd like to look in to that.

File: ClassMoodApp/static/tsc/app.ts [hunk 7]
```diff
@@ -18,6 +18,18 @@ module ClassMoodApp {
         })
         .when("/liveView", {
             templateUrl: "/templates/liveView.html"
+        })
+        .when("/lectureList/:classId", {
+            templateUrl: "templates/lectureList.html",
+            controller: "LectureListController"
+        })
+        .when("/pollingQuestionList/:class_id/:lecture_id", {
```
Comment (nvohra3) [inline ClassMoodApp/static/tsc/app.ts]: Should I just delete all of the entries in this file that relate to template rendering then?

**Review by krabagobanjo — APPROVED** [review]

**Pull request merged**